### ISSUE TITLE
Ax3000 qsdk12.1

### ISSUE THREAD

**Issue by Elkowak** [aoxueos/qsdk_fix#1]
Hi there, i found you built a firmware based on qsdk 12.1 for ax6 on enshan forum, can you try with xiaomi ax3000 ipq5018? i'm trying to build a working firmware using qsdk12.1 for this model but wifi doesn't work properly, kernel driver is loaded, wifi interface shows up but it's broken

**Comment by YumesomeZakura** [comment]
Don't fantasize for 5018, they only give the most of them 256M RAM, that's not enought for qsdk12.1 64bit. And maybe it have some worning in the firmware.

**Comment by Elkowak** [comment]
i know 256mb is insufficient to run 64bit (qualcom sucks with integrated 256mb ram on soc), but what about 32bits, it can be built on 32 bits or qsdk 12.1 only "works" on 64bits? i didn't have access to qsdk release documentation. after found some 32bits owrt qsdk 11.4 for cr8808/cr8806 that boot fine on ax3000 I have hope 12.1 will work too

**Comment by YumesomeZakura** [comment]
Yes, it can.The Cr880X can work on 32bit and Lowmemory mode. you need the ipq50xx_32bit Source，maybe. In the qsdk release documentation, it have a Independent branch called ipq50xx_32bit

**Comment by Elkowak** [comment]
i already have qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source, as i said i built this firmware.
boot: ok
nss: ok
memory free: ok
wifi: doen't work properly, driver load, qca-wifi start but the wireless network is broken for unknown reason, even on luci wpa encryption is missing, maybe the driver is broken, or there is something wrong on ipq5018-mp02.1 dts.

maybe i'm missing something but it's difficulty to work without any documentation

**Comment by YumesomeZakura** [comment]
Don't use the luci, please use the vi to change /etc/config/wireless it have some bug in qsdk's luci. And have you load the right fw.

**Comment by Elkowak** [comment]
just tried again without using luci and editing /etc/config/wireless, wifi network show up with wpa encryption, i connected but the network speed is much slower than expected about 90% slower compared with original firmware, but this is a step foward.

**Comment by YumesomeZakura** [comment]
Maybe, you use a error art or driver. please get art form miwifi or try other driver not os.

**Comment by jlvosloo** [comment]
> i already have qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source, as i said i built this firmware. boot: ok nss: ok memory free: ok wifi: doen't work properly, driver load, qca-wifi start but the wireless network is broken for unknown reason, even on luci wpa encryption is missing, maybe the driver is broken, or there is something wrong on ipq5018-mp02.1 dts.
> 
> maybe i'm missing something but it's difficulty to work without any documentation

Are you able to share the qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source? I would very much appreciate it thanks!

I'm trying to get a IPQ5018 booted.

**Comment by Elkowak** [comment]
> > i already have qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source, as i said i built this firmware. boot: ok nss: ok memory free: ok wifi: doen't work properly, driver load, qca-wifi start but the wireless network is broken for unknown reason, even on luci wpa encryption is missing, maybe the driver is broken, or there is something wrong on ipq5018-mp02.1 dts.
> > maybe i'm missing something but it's difficulty to work without any documentation
> 
> Are you able to share the qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source? I would very much appreciate it thanks!
> 
> I'm trying to get a IPQ5018 booted.

check your email, i sent you the instructions.

**Comment by dm18888** [comment]
> > > i already have qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source, as i said i built this firmware. boot: ok nss: ok memory free: ok wifi: doen't work properly, driver load, qca-wifi start but the wireless network is broken for unknown reason, even on luci wpa encryption is missing, maybe the driver is broken, or there is something wrong on ipq5018-mp02.1 dts.
> > > maybe i'm missing something but it's difficulty to work without any documentation
> > 
> > 
> > Are you able to share the qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source? I would very much appreciate it thanks!
> > I'm trying to get a IPQ5018 booted.
> 
> check your email, i sent you the instructions.

Could you send me a copy ？I would very much appreciate it thanks!
My origin source is 11.5, and ipq5018 works fine, maybe i can  fix the wifi problem.

**Comment by potatoigit** [comment]
> i already have qsdk 12.1 ipq50xx_32 source and qca-networking-2022-spf-12-1_qca_oem source, as i said i built this firmware. boot: ok nss: ok memory free: ok wifi: doen't work properly, driver load, qca-wifi start but the wireless network is broken for unknown reason, even on luci wpa encryption is missing, maybe the driver is broken, or there is something wrong on ipq5018-mp02.1 dts.
> 
> maybe i'm missing something but it's difficulty to work without any documentation

Could you please share the qsdk 12.1 spf source? I have already get a official ipq5018 DR5018 board from a AliBaba store but they refuse to offer any support. I would very much appreciate it!
